v1/core.go

```diff
@@ -64,30 +63,47 @@ type ReBACAdminBackend struct {
 // NewReBACAdminBackend returns a new ReBACAdminBackend instance, configured
 // with given backends.
 func NewReBACAdminBackend(params ReBACAdminBackendParams) (*ReBACAdminBackend, error) {
-	return newReBACAdminBackendWithService(
-		params,
-		newHandlerWithValidation(&handler{
-			Identities:            params.Identities,
-			IdentitiesErrorMapper: params.IdentitiesErrorMapper,
+	// Handlers wrapping one another in this order:
+	//   Dispatcher(Validator(Core))
+	//
+	// Here:
+	// - Dispatcher: determines the availability of the requested HTTP endpoint.
+	// - Validator:  validates the request body/parameters.
+	// - Core:       delegates the control to the service interface implementation.
```

Comment:
Just another idea for the future, these are essentially middlewares in a way. Implementing them here gives us fine grained control over the dispatch/validation of each handler. But if we replaced the dispatcher with middleware you could write less code and check based on the URL path and the `ImplementsX` bools whether to continue or return an HTTP 501 Unimplemented.

Reply:
The reason we're not using standard middlewares and use these stacked handlers, is to avoid magic strings and comparing against string literals (for paths, like `/identities/`). This way, all string-based mappings are done at the auto-generated code, and we just rely on the generated types.